Scripts/modelsystem.py

```diff
@@ -168,52 +178,50 @@ def _add_internal_demand(self, previous_iter_impedance, is_last_iteration):
         log.info("Demand calculation completed")
 
     def _add_external_demand(self,
+                             long_dist_matrices,
                              long_dist_classes = (param.car_classes
                                  + param.long_distance_transit_classes)):
-                                 # + param.freight_classes
-        if not self.ass_model.use_free_flow_speeds:
-            # If we want to assign all trips with traffic congestion,
-            # then add long-distance trips as background demand
-            zone_numbers = self.ass_model.zone_numbers
-            car_matrices = {}
-            try:
-                # Try getting long-distance trips from separate file
-                with self.long_dist_matrices.open(
-                        "demand", "vrk", zone_numbers,
-                        self.mapping, long_dist_classes) as mtx:
-                    log.info("Get matrices from long-distance model run...")
-                    for ass_class in long_dist_classes:
-                        demand = Demand(self.em.purpose, ass_class, mtx[ass_class])
-                        self.dtm.add_demand(demand)
-                        if ass_class in param.car_classes:
-                            car_matrices[ass_class] = demand.matrix
-            except IOError:
-                log.info(
-                    ("Matrices from long-distance model run not found, "
-                     + "getting long trips from base matrices..."))
-                # Long car trips must be in separate file
-                with self.basematrices.open(
-                        "demand", "vrk", zone_numbers,
-                        transport_classes=param.car_classes) as mtx:
-                    for ass_class in param.car_classes:
-                        demand = Demand(self.em.purpose, ass_class, mtx[ass_class])
-                        self.dtm.add_demand(demand)
+        # If we want to assign all trips with traffic congestion,
+        # then add long-distance trips as background demand
+        zone_numbers = self.ass_model.zone_numbers
+        car_matrices = {}
+        if long_dist_matrices is not None:
+            # Try getting long-distance trips from separate file
+            with long_dist_matrices.open(
+                    "demand", "vrk", zone_numbers,
+                    self.mapping, long_dist_classes) as mtx:
```

Comment:
This is the only functional change here, but unfortunately the change of indentation hides it.